gems/actionpack/6.0/actioncontroller.rbs

```diff
@@ -156,3 +156,676 @@ module AbstractController::Callbacks::ClassMethods
+  #
+  # * +permit_all_parameters+ - If it's +true+, all the parameters will be
+  #   permitted by default. The default is +false+.
+  # * +action_on_unpermitted_parameters+ - Allow to control the behavior when parameters
+  #   that are not explicitly permitted are found. The values can be +false+ to just filter them
+  #   out, <tt>:log</tt> to additionally write a message on the logger, or <tt>:raise</tt> to raise
+  #   ActionController::UnpermittedParameters exception. The default value is <tt>:log</tt>
+  #   in test and development environments, +false+ otherwise.
+  #
+  # Examples:
+  #
+  #   params = ActionController::Parameters.new
+  #   params.permitted? # => false
+  #
+  #   ActionController::Parameters.permit_all_parameters = true
+  #
+  #   params = ActionController::Parameters.new
+  #   params.permitted? # => true
+  #
+  #   params = ActionController::Parameters.new(a: "123", b: "456")
+  #   params.permit(:c)
+  #   # => <ActionController::Parameters {} permitted: true>
+  #
+  #   ActionController::Parameters.action_on_unpermitted_parameters = :raise
+  #
+  #   params = ActionController::Parameters.new(a: "123", b: "456")
+  #   params.permit(:c)
+  #   # => ActionController::UnpermittedParameters: found unpermitted keys: a, b
+  #
+  # Please note that these options *are not thread-safe*. In a multi-threaded
+  # environment they should only be set once at boot-time and never mutated at
+  # runtime.
+  #
+  # You can fetch values of <tt>ActionController::Parameters</tt> using either
+  # <tt>:key</tt> or <tt>"key"</tt>.
+  #
+  #   params = ActionController::Parameters.new(key: "value")
+  #   params[:key]  # => "value"
+  #   params["key"] # => "value"
+  class Parameters
+    # Returns a new instance of <tt>ActionController::Parameters</tt>.
+    # Also, sets the +permitted+ attribute to the default value of
+    # <tt>ActionController::Parameters.permit_all_parameters</tt>.
+    #
+    #   class Person < ActiveRecord::Base
+    #   end
+    #
+    #   params = ActionController::Parameters.new(name: "Francesco")
+    #   params.permitted?  # => false
+    #   Person.new(params) # => ActiveModel::ForbiddenAttributesError
+    #
+    #   ActionController::Parameters.permit_all_parameters = true
+    #
+    #   params = ActionController::Parameters.new(name: "Francesco")
+    #   params.permitted?  # => true
+    #   Person.new(params) # => #<Person id: nil, name: "Francesco">
+    def initialize: (?::Hash[untyped, untyped] parameters) -> untyped
+
+    def keys: () -> Array[untyped]
+    def key?: () -> bool
+    def has_key?: () -> bool
+    def values: () -> Array[untyped]
+    def has_value?: () -> bool
+    def empty?: () -> bool
+    def include?: (untyped) -> bool
+    def as_json: () -> String
+    def to_s: () -> String
+    def each_key: () { (untyped) -> untyped} -> Hash[untyped, untyped]
+                | () -> Enumerator[untyped, self]
```

Comment:
I added these methods.

Reply:
Why didn't you use open classes?
The other changes seem redundant.

Reply:
I think this is not a patch because the original implementation is defined in the same file. So it's better to add these definitions to the same RBS file. Of course, I'm okay with using the open class way.

Reply:
I am talking about moving from gems/actionpack/6.0/actionpack-generated.rbs to gems/actionpack/6.0/actioncontroller.rbs.

Reply:
Yes, I am also talking about it. But it seems using open class is better. Okay, I update it from now on.